staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -159,16 +159,18 @@ func (i *indexedWatchers) terminateAll(objectType reflect.Type, done func(*cache
 // if you set fire time at X, you can get the bookmark within (X-1,X+1) period.
 type watcherBookmarkTimeBuckets struct {
 	lock              sync.Mutex
-	watchersBuckets   map[int64][]*cacheWatcher
-	startBucketID     int64
+	watchersBuckets   map[time.Duration][]*cacheWatcher
```

Comment:
suggest leaving the key as int64 to avoid confusion about granularity of the map keys

Reply:
since we're only storing values on second granularity, I think making the key a duration causes 1,000,000,000 empty lookups as we're iterating from one second to the next

Reply:
also add a comment that the watchersBucket key is the number of seconds since `createTime`